dist/tools/doccheck/check.sh

```diff
@@ -17,6 +19,23 @@ then
     echo "ERROR: Doxygen generates the following warnings:"
     echo "${ERRORS}"
     exit 2
-else
-    exit 0
 fi
+
+# Check all groups are defined
+
+DEFINED_GROUPS="$(git grep defgroup | grep -v vendor | grep -oE '@defgroup[ ]+[^ ]+' | grep -oE '[^ ]+$' | sort -u)"
+
+UNDEFINED_GROUPS=$( \
+    for group in $(git grep '@ingroup' | grep -v vendor | grep -oE '[^ ]+$' | sort -u); \
+    do \
+        echo "$DEFINED_GROUPS" | grep -xq "$group" || echo "$group"; \
```

Comment:
Might it be possible to also include the filename here?

Reply:
I can look at that. Not sure how complex it is.